tests/testserverclient.nim

```diff
@@ -49,9 +49,51 @@ suite "Socket Server/Client RPC":
     except CatchableError as e:
       check e.msg == """{"code":-32001,"message":"Unknown payload"}"""
 
+  test "Client close and isConnected":
+    check client.isConnected() == true
+    # Is socket server close broken?
+    # waitFor client.close()
```

Comment:
I didn't spend long looking at it. Do you have a suggestion/guidance on how I can fix it? If it isn't trivial, then I'll create an issue for it.

Reply:
I've just pushed a fix which solves the problem in the tests. Can you review the change when you get a chance? I'm not sure if it might effect something else.